### DISCUSSION TITLE
Setting up with Django auth

### DISCUSSION THREAD

**Discussion by ughstudios** [processone/ejabberd#3760] [Questions] [answered]
I found this script on the website, but I'm not sure how to use it. I have a django application using Django-rest-framework and Django. I want people to be able to login to the xmpp server with their django account. Is this possible?

https://www.ejabberd.im/files/contributions/ejabberd_auth_bridge.py.txt

**Comment by badlop** [comment]
I guess you found this script when looking for external authentication scripts: https://docs.ejabberd.im/admin/configuration/authentication/#external-script
which mentions https://www.ejabberd.im/extauth/
which lists https://www.ejabberd.im/node/4000/index.html

In that case, tell ejabberd to use that script for extauth, something like:
```yaml
auth_method: [external]
extauth_program: /etc/ejabberd/ejabberd_auth_bridge.py
```
As the script is very old, quite probably the python code will not work with recent python versions. Regarding the ejabberd part, extauth in ejabberd has not changed in years, so that should not be a problem.

**Reply by badlop** [reply]
Related: https://stackoverflow.com/questions/70872369/what-does-ejabberd-authentication-do

**Comment by ughstudios** [comment, marked as answer]
@badlop It looks like if you use an external auth script, the first password you accept from your script becomes that users permanent password. Is there a way to change this?

**Reply by badlop** [reply]
You found the explanation: `auth_use_cache` is enabled by default. For reference: https://github.com/processone/ejabberd/discussions/3768#discussioncomment-2153064
